sdf/concepts/custom-serialization.mdx

```diff
@@ -0,0 +1,200 @@
+      human-label:
+        type: string
+  visibility:
+    type: enum
+    oneOf:
+      public:
+        type: null
+      private:
+        type: null
+```
+
+A valid example of a record that could be deserialized properly for the `new` type, would be something like:
+
+```yaml
+{
+    "title": "Fluvio is great!",
+    "image_url": "https://fluvio.io/img/fluvio.jpg"
+    "keywords": [
+        {
+            "slug": "tech",
+            "human-label": "Technology"
+        }
+    ],
+    "visibility": "Public"
+}
+```
+
+#### Json custom deserialization
+
+In some occasions, we may want to use different casing. For that reason, we have the `serialize` and `deserialize` configuration on object attributes and enum variants.
+
+The serialize and deserialize configurations facilitate a flexible and robust way to manage data transformations. By clearly defining how properties should be renamed during these processes, the system can seamlessly integrate with external APIs or services while maintaining an internal structure that is coherent and easy to manage.
+
+##### Serialization configuration
+
+The serialize configuration is used to specify how a property should be transformed during the serialization process. It is defined within the properties section of an object type or within the variants of an enum and can have the following options:
+- `rename`: Specifies the new name for the property when serialized.
+
+Example:
+```yaml
+types:
+  new:
+    type: object
+    properties:
+      image-url:
+        type: string
+        optional: true
+        serialize:
+          rename: imageUrl
+```
+
+In this example, the image-url property will be serialized as imageUrl. The same configuration could be applied on enum variants.
+
+
+##### Deserialization configuration
+
+The deserialize configuration is used to specify how a property should be transformed during the deserialization process. It is also defined within the properties section of an object type or within the variants of an enum and can have the following options:
+
+- `rename`: Specifies the new name for the property when deserialized.
+
+```
+types:
+  visibility:
+    type: enum
+    oneOf:
+      public:
+        deserialize:
+          rename: private
+        type: null
+      private:
+        deserialize:
+          rename: private
+        type: null
+```
+In this example, the `visibility` type will successfully deserialize when the input is `public` or `private`. The same configuration could be applied on object properties.
+
+#### Custom deserialization in example
+
+For instance, if we want to parse the following data:
+
+```yaml
+{
+    "title": "Fluvio is great!",
+    "imageUrl": "https://fluvio.io/img/fluvio.jpg"
+    "keywords": [
+        {
+            "slug": "tech",
+            "humanLabel": "Technology"
+        }
+    ],
+    "visibility": "public"
+}
+```
+
+We could define the types at this way, to properly deserialize and serialize using that format:
```

Comment:
We could define the types at this way, to properly deserialize and serialize using that format: ->
Let's look at the full example